output/summary/data.go

```diff
@@ -0,0 +1,367 @@
+type aggregatedMetric struct {
+	// FIXME (@joan): Drop this and replace it with a concrete copy of the metric data we want to track
+	// to avoid any potential confusion.
+	Metric *metrics.Metric
+
+	// FIXME (@joan): Introduce our own way of tracking thresholds, and whether they're crossed or not.
+	// Without relying on the internal submetrics the engine maintains specifically for thresholds.
+	// Thresholds []OurThreshold // { crossed: boolean }
+
+	Sink metrics.Sink
+}
+
+func newAggregatedMetric(metric *metrics.Metric) aggregatedMetric {
+	return aggregatedMetric{
+		Metric: metric,
+		Sink:   metrics.NewSink(metric.Type),
+	}
+}
+
+type aggregatedChecksData struct {
+	checks        map[string]*lib.Check
+	orderedChecks []*lib.Check
+}
+
+func newAggregatedChecksData() *aggregatedChecksData {
+	return &aggregatedChecksData{
+		checks:        make(map[string]*lib.Check),
+		orderedChecks: make([]*lib.Check, 0),
+	}
+}
+
+func (a *aggregatedChecksData) checkFor(name string) *lib.Check {
+	check, ok := a.checks[name]
+	if !ok {
+		var err error
+		check, err = lib.NewCheck(name, &lib.Group{}) // FIXME: Do we really need the group?
+		if err != nil {
+			panic(err) // This should never happen
+		}
+		a.checks[name] = check
+		a.orderedChecks = append(a.orderedChecks, check)
+	}
+	return check
+}
+
+func populateSummaryGroup(
+	summaryMode lib.SummaryMode,
+	summaryGroup *lib.SummaryGroup,
+	groupData aggregatedGroupData,
+	testRunDuration time.Duration,
+	summaryTrendStats []string,
+) {
+	// First, we populate the checks metrics, which are treated independently.
+	populateSummaryChecks(summaryGroup, groupData, testRunDuration, summaryTrendStats)
+
+	// Then, we store the metrics.
+	storeMetric := func(
+		dest lib.SummaryMetrics,
+		info lib.SummaryMetricInfo,
+		sink metrics.Sink,
+		testDuration time.Duration,
+		summaryTrendStats []string,
+	) {
+		summaryMetric := lib.NewSummaryMetricFrom(info, sink, testDuration, summaryTrendStats)
+
+		switch {
+		case isSkippedMetric(summaryMode, info.Name):
+			// Do nothing, just skip.
+		case isHTTPMetric(info.Name):
+			dest.HTTP[info.Name] = summaryMetric
+		case isExecutionMetric(info.Name):
+			dest.Execution[info.Name] = summaryMetric
+		case isNetworkMetric(info.Name):
+			dest.Network[info.Name] = summaryMetric
+		case isBrowserMetric(info.Name):
+			dest.Browser[info.Name] = summaryMetric
+		case isGrpcMetric(info.Name):
+			dest.Grpc[info.Name] = summaryMetric
+		case isWebSocketsMetric(info.Name):
+			dest.WebSocket[info.Name] = summaryMetric
+		case isWebVitalsMetric(info.Name):
+			dest.WebVitals[info.Name] = summaryMetric
+		default:
+			dest.Custom[info.Name] = summaryMetric
+		}
+	}
+
+	for _, metricData := range groupData.aggregatedMetrics {
+		storeMetric(
+			summaryGroup.Metrics,
+			lib.SummaryMetricInfo{
+				Name:     metricData.Metric.Name,
+				Type:     metricData.Metric.Type.String(),
+				Contains: metricData.Metric.Contains.String(),
+			},
+			metricData.Sink,
```

Comment:
I am not particularly certain this should actually get the whole `Metric` instead of just some parts of it

Reply:
Do you mean that `aggregatedMetric` may already hold the `lib.SummaryMetricInfo`, which is a subset of the data hold by the `*metrics.Metric`, and replace that instead? 

If so, I agree, and I'm happy to make that change. 
But, could you confirm it, please? If not, please can you clarify what do you mean? 🙏🏻

Reply:
Yeah - basically `metrics.Metric` will get thinned out, so a lot of the stuff you currently don't use from it will be removed. So no real reason to copy stuff around IMO.